File: lib/layer2_core/light/e_inverse_primary_light.dart
import '../../layer1_raw/design_token.dart';
import '../../layer1_raw/light/system.dart';
import '../../layer1_raw/light/standard.dart';
import '../../layer1_raw/light/blue.dart';
import '../../layer1_raw/light/green.dart';
import '../../layer1_raw/light/red.dart';
import '../../layer1_raw/light/nothing.dart';

class EInversePrimaryLight {
  static DesignToken getToken(dynamic activeTheme) {
    switch (activeTheme.toString().split('.').last) {
      case 'system': return csyl24;
      case 'standard': return cstl24;
      case 'blue': return cbl24;
      case 'green': return cgl24;
      case 'red': return crl24;
      case 'nothing': return cnol24;
      default: return cstl24;
    }
  }
}

File: android/src/main/kotlin/com/example/design_engine/layer2_core/light/EInversePrimaryLight.kt
package com.example.design_engine.layer2_core.light
import com.example.design_engine.R
import com.example.design_engine.layer3_logic.AppTheme
object EInversePrimaryLight {
    fun getResId(activeTheme: AppTheme): Int {
        return when (activeTheme) {
            AppTheme.SYSTEM -> R.string.csyl24
            AppTheme.STANDARD -> R.color.cstl24
            AppTheme.BLUE -> R.color.cbl24
            AppTheme.GREEN -> R.color.cgl24
            AppTheme.RED -> R.color.crl24
            AppTheme.NOTHING -> R.color.cnol24
        }
    }
}
